### PR TITLE
Use WARN_CFLAGS which are only set with --enable-compiler-warnings

### PR DESCRIPTION
**Pull request by rbuj**

### CODE REVIEW

**Comment by sc0w** [comment]
@mate-desktop/core-team 

Personally, I prefer to build over master branch of mate-common directly, see https://github.com/mate-desktop/mate-notification-daemon/pull/161

**Comment by raveit65** [comment]
Works fine, ecxept of mix usage of tabs and spaces in config summary.
```
[rave@mother mate-notification-daemon-dist]$ ./autogen.sh 

        compiler:                 gcc
	cflags:		          -g -O2
        warning flags:            -Wall -Wmissing-prototypes

[rave@mother mate-notification-daemon-dist]$ ./configure --enable-compile-warnings=no

        compiler:                 gcc
	cflags:		          -g -O2
        warning flags:            

[rave@mother mate-notification-daemon-dist]$ ./configure --enable-compile-warnings=yes

        compiler:                 gcc
	cflags:		          -g -O2
        warning flags:            -Wall -Wmissing-prototypes

[rave@mother mate-notification-daemon-dist]$ ./configure --enable-compile-warnings=maximum

        compiler:                 gcc
	cflags:		          -g -O2
        warning flags:            -Wall -Wmissing-prototypes -Wbad-function-cast -Wcast-align -Wextra -Wformat-nonliteral -Wmissing-declarations -Wmissing-field-initializers -Wnested-externs -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-prototypes -Wno-sign-compare

[rave@mother mate-notification-daemon-dist]$ ./configure --enable-compile-warnings=error

			mate-notification-daemon 1.23.0
			================================

	prefix:                   /usr/local
	exec_prefix:              ${prefix}
	bindir:                   ${exec_prefix}/bin
	datadir:                  ${datarootdir}
	source code location:     .
        compiler:                 gcc
	cflags:		          -g -O2
        warning flags:            -Wall -Wmissing-prototypes -Wbad-function-cast -Wcast-align -Wextra -Wformat-nonliteral -Wmissing-declarations -Wmissing-field-initializers -Wnested-externs -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-prototypes -Wno-sign-compare -Werror

	dbus-1 system.d           /usr/local/etc/dbus-1/system.d
	dbus-1 services           ${prefix}/share/dbus-1/services

[rave@mother mate-notification-daemon-dist]$ 
```
@rbuj 
Can you please fix that?

mate-common-1.23.3 has everything what we need for this change.
And it will reminds us to remove that from travis.yml in future when all distros use MATE-1.24.
I don't think that someone wants to add more changes mate-common ***now*** or i near future.
So building from 1.23.3 is sufficient.......IHMO.

File: configure.ac [hunk 1]
```diff
@@ -255,14 +211,16 @@ AC_OUTPUT
...
-			=========================
+			================================
 
 	prefix:                   ${prefix}
 	exec_prefix:              ${exec_prefix}
 	bindir:                   ${bindir}
 	datadir:                  ${datadir}
 	source code location:     ${srcdir}
-	cflags:		              ${AM_CFLAGS}
+        compiler:                 ${CC}
```
Comment (raveit65) [inline configure.ac]: mix usage of tabs an spaces.

Comment (rbuj) [reply configure.ac]: @raveit65 fixed

**Comment by sc0w** [comment]
why I see in https://github.com/mate-desktop/mate-notification-daemon/pull/161 more warnings and here less warnings?

for example, this warning is missing here:

`daemon.c:682:72: warning: unused parameter 'event' [-Wunused-parameter]`

**Comment by rbuj** [comment]
@sc0w I fixed it before that you edited your previous comment. Anyway, thanks for the detailed log ;-)
The autogen.sh compiler warning flags are overridden in configure step.

**Pull request merged**